### PR TITLE
Shopping cart

### PR DESCRIPTION
**Pull request by sfoa6558**

### CODE REVIEW

File: shopping_cart/features/add_coupon.feature [hunk 1]
```diff
@@ -0,0 +1,24 @@
+Feature:Adding coupons to my order
```
Comment (zbright) [inline shopping_cart/features/add_coupon.feature]: You should consider adding more scenarios here. The idea of an invalid coupon expands beyond exipired. What if they enter an invalid coupon code, or conditions for a valid code are not met?

File: shopping_cart/features/add_remove_items.feature [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    Then I should see an iphone in my shopping cart
+
+
+
+  Scenario Outline: adding multiple items to Shopping Cart
+    Given I am on the shopping cart page
+    When I add <number> different itemss in my shopping cart
+    Then I should see the same <number> different items in my shopping cart
+
+    Examples:
```
Comment (zbright) [inline shopping_cart/features/add_remove_items.feature]: This is a cool idea. I dig it

File: shopping_cart/features/add_remove_items.feature [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      |   2    |
+      |   3    |
+      |   4    |
+      |   5    |
+
+
+  Scenario: adding an item with a quantity of 0 to a Shopping Cart
+    Given I am on the shopping cart page
+    When I change the quantity of an iphone to 0
+    And  attempt to add the iphone to my shopping cart
```
Comment (zbright) [inline shopping_cart/features/add_remove_items.feature]: If you are already on the cart page, why would there be a button to add the item? Do you mean to just update the quantity, because in that case the item should just get removed from the cart

File: shopping_cart/features/add_remove_items.feature [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+    Examples:
+      | number |
+      |   2    |
+      |   3    |
+      |   4    |
+      |   5    |
+
+
+   Scenario: removing more items than in Shopping Cart
```
Comment (zbright) [inline shopping_cart/features/add_remove_items.feature]: This is pretty ambiguous. If you reword it to something like `removing an item that is not in the cart` it might make more sense. Removing the 20 valid items is not relevant to this test

File: shopping_cart/features/item_cart_page.feature [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    Given I am on the item page
+    When I click on the shopping cart link
+    Then I am taken back to the shopping cart
+
+   Scenario: verifying items still exist in the shopping cart page
+     Given I am on the item page
+     When I navigate back to the shopping cart link
+     Then I see all the items that were there previously
+
+   Scenario: unable to navigate to the item page
```
Comment (zbright) [inline shopping_cart/features/item_cart_page.feature]: Am I missing something, or is this test essentially ensuring that if the user does nothing then nothing happens?

File: shopping_cart/features/item_cart_page.feature [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    Given I am on the shopping cart page
+    When I click on the title of my stereo
+    Then I should be taken to the item page for that stereo
+
+  Scenario: navigate to the item page by clicking the picture  of the item
+    Given I am on the shopping cart page
+    When I click on the picture of my xbox
+    Then I should be taken to the item page for that xbox
+
+
```
Comment (zbright) [inline shopping_cart/features/item_cart_page.feature]: extra space

File: shopping_cart/features/item_cart_page.feature [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+   Scenario: verifying items still exist in the shopping cart page
+     Given I am on the item page
+     When I navigate back to the shopping cart link
+     Then I see all the items that were there previously
+
+   Scenario: unable to navigate to the item page
+     Given I am on the shopping cart page
+     When I do not click on a picture or title of an item
+     Then I should not be taken to the item page for that item
+
```
Comment (zbright) [inline shopping_cart/features/item_cart_page.feature]: extra spaces

File: shopping_cart/features/login.feature [hunk 8]
```diff
@@ -0,0 +1,38 @@
+Feature: Login shopping cart
```
Comment (zbright) [inline shopping_cart/features/login.feature]: Well done on this file

File: shopping_cart/features/add_remove_items.feature [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+  Scenario: adding an item to a Shopping Cart
+    Given I am on the shopping cart page
+    When I add an iphone to my shopping cart
+    Then I should see an iphone in my shopping cart
+
+
+
+  Scenario Outline: adding multiple items to Shopping Cart
+    Given I am on the shopping cart page
+    When I add <number> different itemss in my shopping cart
```
Comment (djkotowski) [inline shopping_cart/features/add_remove_items.feature]: Nitpick: spelling

**Comment by djkotowski** [comment]
Nice work!  Just a few comments throughout.  No need for a resubmission.
